### ISSUE TITLE
Using initializeWithKey with ES7 Decorators

### ISSUE THREAD

**Issue by qnm** [erikras/redux-form#38]
Is it possible to use `initializeWithKey` alongside the ES7 decorator?

It looks like it's possible to do, but my react-fu isn't sufficient to understand how to achieve it.

**Comment by erikras** [comment]
I'm not sure what you mean by "alongside". Explain what you're trying to achieve.

**Comment by qnm** [comment]
@erikras I'd like to use `initializeWithKey` with `@connectReduxForm`. Is that possible?

**Comment by erikras** [comment]
Can you describe your use case a bit more? I'm still not understanding.

You mean you want to _pass initial values_ to `@connectReduxForm`?

**Comment by qnm** [comment]
@erikras Sorry.

In order to use `formKey`, the following is applied to the `QueryForm` component class:.

```
function mapStateToProps(state) {
  return { registration: state.form.registration };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ initializeWithKey }, dispatch);
}

// apply connect() to bind it to Redux state
QueryForm = connect(mapStateToProps, mapDispatchToProps)(QueryForm);

// export the wrapped component
export default QueryForm;
```

Is it possible to achieve the same outcome using `@connectReduxForm`?

**Comment by erikras** [comment]
Let me rephrase to make sure I'm understanding. You want `initializeWithKey` to be passed in as a `prop`, [like `initializeForm` is](https://github.com/erikras/redux-form#-initializeformdataobject--function)?

**Comment by erikras** [comment]
Btw, you are correct in how you get access to `initializeWithKey` currently.

**Comment by qnm** [comment]
Passing in via a prop (a default prop?) would be good. I'm looking for a more succinct syntax, really.

**Comment by erikras** [comment]
Where are you calling it from? Passing `initializeWithKey()` in as a prop (already bound to dispatch and the form itself) would be pretty simple.

But you got me thinking, that I wonder if a decorated form component that accepted a `formKey` prop might also should accept an `initialValues` prop, and it could call `initializeWithKey()` on `componentWillMount()`, for example.

**Comment by qnm** [comment]
> Where are you calling it from? Passing initializeWithKey() in as a prop (already bound to dispatch and the form itself) would be pretty simple.

I'm calling it from inside `componentWillMount()`.

How would I pass `initializeWithKey` in as a prop?

> But you got me thinking, that I wonder if a decorated form component that accepted a formKey prop might also should accept an initialValues prop, and it could call initializeWithKey() on componentWillMount(), for example.

That sounds good!

**Comment by erikras** [comment]
No, `@connectReduxForm` would give it to your component as a prop. So you'd just do:
```javascript
componentWillMount() {
  this.props.initializeWithKey(this.props.initialValues);
}
```
However, that's the exact function that I was suggesting that redux-form also provide.

Thanks for the thought-provoking input.

**Comment by qnm** [comment]
Ah, of course. That would be very helpful!

Thanks for the great library, @erikras!

**Comment by erikras** [comment]
Implemented `initialValues` prop in [v1.2.0](https://github.com/erikras/redux-form/releases/tag/v1.2.0).
